File: app/src/main/java/com/brayanbedritchuk/zerotohero/view/exercise_list/presenter/ExerciseListViewModel.java
package com.brayanbedritchuk.zerotohero.view.exercise_list.presenter;

import com.brayanbedritchuk.zerotohero.base.BaseViewModel;
import com.brayanbedritchuk.zerotohero.model.Exercise;

import java.util.ArrayList;
import java.util.List;

public class ExerciseListViewModel extends BaseViewModel {

    private final List<Exercise> exercises;

    public ExerciseListViewModel() {
        this.exercises = new ArrayList<>();
    }

    public List<Exercise> getExercises() {
        return exercises;
    }
}

File: shared/src/androidMain/kotlin/br/com/sailboat/zerotohero/presentation/ExerciseListViewModel.kt
package br.com.sailboat.zerotohero.presentation

import androidx.lifecycle.viewModelScope
import br.com.sailboat.zerotohero.base.BaseViewModel
import br.com.sailboat.zerotohero.domain.usecase.GetExercises
import kotlinx.coroutines.launch

internal class ExerciseListViewModel(
    override val viewState: ExerciseListViewState = ExerciseListViewState(),
    private val getExercises: GetExercises,
) : BaseViewModel<ExerciseListViewState, ExerciseListViewAction>() {

    override fun dispatchViewAction(viewAction: ExerciseListViewAction) {
        when (viewAction) {
            ExerciseListViewAction.OnStart -> onStart()
        }
    }

    private fun onStart() = viewModelScope.launch {
        val exercises = getExercises()
        viewState.exerciseList.value = exercises
    }
}
